### DISCUSSION TITLE
스터디 참여, 탈퇴 API 명세에 대해

### DISCUSSION THREAD

**Discussion by sc0116** [woowacourse-teams/2022-moamoa#263] [토론]
안녕하세요 저 짱군데요, 
스터디원 탈퇴 기능을 구현하려고 하다가 든 고민이 있어 질문남겨용.

현재 스터디원 참여 API가 `POST /studies/{study-id}`인데여(`/api/`는 편의상 생략합니다 반박시 직접 수정해서 붙여주세요).
그래서 스터디원 탈퇴를 `DELETE /studies/{study-id}`로 하려다가 뭔가 어색하더라구요!

API는 정보의 자원을 표현하는거로 알고있는데요, `/studies/{study-id}`가 표현하는 자원은 스터디겠죠?
그렇다면 `DELETE /studies/{study-id}`는 스터디 삭제가 되어야 한다고 생각했구요.
이런 관점에서 봤을 때 스터디 참여 API가 `POST /studies/{study-id}`인 것도 어색하다고 생각했어요! (물론 스터디 생성은 `POST /studies`라서 겹치진 않네요!)

그래서 다음과 같이 API를 수정하면 어떨까 생각해봤는데 다른 팀원분들은 어떠신가요?
- 스터디 참여: `POST /studies/{study-id}/join`, `POST /studies/{study-id}/participate?participation?` 등
- 스터디 탈퇴: `DELETE /studies/{study-id}/leave` 등

@jaejae-yoo @tco0427 @wilgur513 @airman5573 @nan-noo

**Comment by nan-noo** [comment]
스터디원 탈퇴
`DELETE studies/{study-id}/members/{member-id}`
스터디 참여
`POST studies/{study-id}/members`

요런 식으로 다른 스터디 커뮤니티, 후기, 링크 생성처럼 바꾸면 좋을 것 같아요

**Reply by sc0116** [reply]
- 스터디 참여: `DELETE /studies/{study-id}/participants`
- 스터디 탈퇴: `DELETE /studies/{study-id}/participants`
요건 어떤가요? (participants는 members로 바꿔도 상관없음)

어차피 토큰으로 인가를 하니까 `/{member-id}`는 안해도 될 것 같다는 생각이 드네요!

**Reply by tco0427** [reply]
‘참여자들’ 이라는 단어가 괜찮은것 같네요
참여자를 create한다는 참여한다는 의미가 전달될 것 같고, 참여자를 delete한다 는 탈퇴라는 의미가 전달되는 것 같습니다!

**Comment by jaejae-yoo** [comment]
회원 관련 API에서 members를 사용하고 있어서, 

스터디원 탈퇴 `DELETE studies/{study-id}/members`
스터디 참여 `POST studies/{study-id}/members`

로 해도 좋을 것 같아요
